src/Test/EntityManagerMockFactory.php

```diff
@@ -19,8 +19,9 @@
 use Doctrine\ORM\EntityManagerInterface;
 use Doctrine\ORM\EntityRepository;
 use Doctrine\ORM\QueryBuilder;
+use PHPUnit\Framework\TestCase;
 
-class EntityManagerMockFactory
+class EntityManagerMockFactory extends TestCase
```

Comment:
Why? You should not extend this class

Reply:
maybe but when i done this modification i do not have issue on ecommerce test anymore.

can you take a look and explain how to solve issue on EntityManagerMockFactory i'm going crazy on this. something change from 2 weeks all tests fails

https://github.com/sonata-project/ecommerce/pull/631

Reply:
Maybe it's better to create an `EntityManagerMockFactoryTrait` which can be used inside the test. So you can use multiple extension points for your test.

Reply:
no changes ! 

I have created trait like this 

```
trait EntityManagerMockFactoryTrait
{
    /**
     * @return EntityManagerInterface
     */
    public static function create(\PHPUnit\Framework\TestCase $test, \Closure $qbCallback, $fields)
    {
        $query = $test->createMock(AbstractQuery::class);
        $query->method('execute')->willReturn(true);

        $qb = $test->createMock(QueryBuilder::class);

        $qb->method('select')->willReturn($qb);
        $qb->method('getQuery')->willReturn($query);
        $qb->method('where')->willReturn($qb);
        $qb->method('orderBy')->willReturn($qb);
        $qb->method('andWhere')->willReturn($qb);
        $qb->method('leftJoin')->willReturn($qb);

        $qbCallback($qb);

        $repository = $test->createMock(EntityRepository::class);
        $repository->method('createQueryBuilder')->willReturn($qb);

        $metadata = $test->createMock(ClassMetadata::class);
        $metadata->method('getFieldNames')->willReturn($fields);
        $metadata->method('getName')->willReturn('className');

        $em = $test->createMock(EntityManager::class);
        $em->method('getRepository')->willReturn($repository);
        $em->method('getClassMetadata')->willReturn($metadata);

        return $em;
    }
}

and use this trait into test files

```
class InvoiceManagerTest extends TestCase
{
    use EntityManagerMockFactoryTrait;
    (...)

}
```
Same issue error 

```
Error: Call to protected method PHPUnit\Framework\TestCase::createMock() from context 'Sonata\Doctrine\Test\EntityManagerMockFactoryTrait'

C:\wamp64\www\ecommerce\vendor\sonata-project\doctrine-extensions\src\Test\EntityManagerMockFactoryTrait.php:30
C:\wamp64\www\ecommerce\tests\ProductBundle\Entity\ProductManagerTest.php:110
C:\wamp64\www\ecommerce\tests\ProductBundle\Entity\ProductManagerTest.php:103

```

Works great when just extend EntityManagerMockFactory